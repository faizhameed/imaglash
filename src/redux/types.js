const actionTypes = {
  UNSPLASH_INITIALISED: "UNSPLASH_INITIALISED",
  ON_PAGE_LOAD: "ON_PAGE_LOAD",
  SEARCH_QUERY_CHANGED: "SEARCH_QUERY_CHANGED",
  LOAD_MORE_DATA: "lOAD_MORE_DATA",
  RANDOM_IMAGE_TRIGGER: "RANDOM_IMAGE_TRIGGER",
  USER_CLICKS_IMAGE: "USER_CLICKS_IMAGE",
};

export default actionTypes;
