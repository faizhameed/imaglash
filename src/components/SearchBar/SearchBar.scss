$yellow: #f89f2e;
$secondaryColor: #49c4c0;

.tag-container {
  display: flex;
  flex-wrap: wrap;
  p {
    margin-left: 10px;
    color: $yellow;
    font-weight: bold;
    cursor: pointer;
  }
}
.search-input {
  padding: 1em;
  margin: 0.5em 0;
  color: #222222;
  background: white;
  border: none;
  border-radius: 3px;
  width: 50%;
  min-width: 200px;
  height: 25px;
}
.inp-btn-cont {
  display: flex;
  align-items: center;
  .icon-btn {
    width: 40px;
    height: 40px;
    border-radius: 50%;
    background-color: $secondaryColor;
    transform: translateX(-20px);
    transition: 0.1s;
    border: none;
    outline: none;
    cursor: pointer;
    &:hover {
      background-color: #222222;
    }
    &:active {
      transform: translateX(-25px);
    }
    svg {
      margin: auto;
      width: 20px;
      height: auto;
    }
  }
}
