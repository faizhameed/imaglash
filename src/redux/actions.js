import actionTypes from "./types";

import Unsplash, { toJson } from "unsplash-js";

const unsplash = new Unsplash({
  accessKey: "C6bzEnmvODyk9wJ3ig0V4E7p9pwGVkZNejOftxLLdVI",
});

export const changeSearchQuery = (query) => {
  return function (dispatch) {
    if (unsplash) {
      unsplash.search
        .photos(query, 1, 9)
        .then(toJson)
        .then((json) => {
          dispatch({
            type: actionTypes.SEARCH_QUERY_CHANGED,
            payload: json,
            query: query,
          });
        });
    }
  };
};

export const updatePageNumber = (page, query) => {
  return function (dispatch) {
    if (unsplash) {
      unsplash.search
        .photos(query, page + 1, 9)
        .then(toJson)
        .then((json) => {
          dispatch({
            type: actionTypes.LOAD_MORE_DATA,
            payload: json,
            query,
          });
        });
    }
  };
};

export const setCollection = () => {
  return function (dispatch) {
    if (unsplash) {
      unsplash.search
        .photos("book", 1, 9)
        .then(toJson)
        .then((json) => {
          dispatch({
            type: actionTypes.ON_PAGE_LOAD,
            payload: json,
          });
        });
    }
  };
};
