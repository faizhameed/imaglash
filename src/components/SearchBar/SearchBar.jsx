import React from "react";
import SearchIcon from "../../assets/svgs/search.svg";

import { connect } from "react-redux";
import { changeSearchQuery } from "../../redux/actions";

import "./SearchBar.scss";

const SearchBar = ({ tags, changeSearchQuery }) => {
  let inputQuery = React.createRef();
  const handleSearch = () => {
    if (inputQuery.current.value) changeSearchQuery(inputQuery.current.value);
  };
  return (
    <div>
      <div className="tag-container">
        <h4>Search by Tags:</h4>
        {tags.map((tag, i) => (
          <p onClick={() => changeSearchQuery(tag)} key={i}>
            {tag}
          </p>
        ))}
      </div>
      <div className="inp-btn-cont">
        <input
          ref={inputQuery}
          className="search-input"
          type="text"
          placeholder="Search for images here..."
          onKeyPress={(ev) => {
            if (ev.key === "Enter") {
              handleSearch();
            }
          }}
        />
        <button className="icon-btn" onClick={handleSearch}>
          <SearchIcon />
        </button>
      </div>
    </div>
  );
};

const mapStateToProps = ({ unsplashReducer: { tags } }) => ({
  tags,
});

const mapDispatchToProps = (dispatch) => ({
  changeSearchQuery: (data) => dispatch(changeSearchQuery(data)),
});

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);
