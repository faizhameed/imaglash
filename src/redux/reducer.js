import actionTypes from "./types";
import { combineReducers } from "redux";

const UNSPLASH_STATE = {
  unsplash: null,
  tags: [
    "Dog",
    "Cat",
    "Space",
    "Nature",
    "Business",
    "Office",
    "Coffee",
    "World",
    "Wildlife",
    "Beach",
    "Digital",
    "Meeting",
    "Cars",
    "Games",
    "Holiday",
  ],
  currentQuery: "Book",
  page: 1,
  collection: [],
  totalPages: 0,
  randomImageUrl: "",
  userImageSelection: null,
};

export function unsplashReducer(state = UNSPLASH_STATE, action) {
  switch (action.type) {
    case actionTypes.UNSPLASH_INITIALISED:
      return {
        ...state,
        unsplash: action.payload,
      };
    case actionTypes.ON_PAGE_LOAD:
      return {
        ...state,
        collection: [...action.payload.results],
        totalPages: action.payload.total_pages,
        page: 1,
      };

    case actionTypes.SEARCH_QUERY_CHANGED:
      return {
        ...state,

        collection: [...action.payload.results],
        page: 1,
        totalPages: action.payload.total_pages,
        currentQuery: action.query,
      };

    case actionTypes.LOAD_MORE_DATA:
      if (state.page < state.totalPages)
        return {
          ...state,
          page: state.page + 1,
          collection: [...state.collection, ...action.payload.results],
        };
      break;
    case actionTypes.RANDOM_IMAGE_TRIGGER:
      return {
        ...state,
        randomImageUrl: action.payload,
      };

    case actionTypes.USER_CLICKS_IMAGE:
      return {
        ...state,
        userImageSelection: action.payload,
      };
    default:
      return state;
  }
}

const rootReducer = combineReducers({
  unsplashReducer,
});

export default rootReducer;
