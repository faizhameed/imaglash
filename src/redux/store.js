import { createStore } from "redux";
import rootReducer from "./reducer";
import { applyMiddleware } from "redux";
import { createLogger } from "redux-logger";
import thunkMiddleware from "redux-thunk";

const logger = createLogger();
/**
 * we are creating global store and using the store in our root file to have
 * all components to access it
 *
 */

const middlewares = [thunkMiddleware];
/**
 * we need not run logger in production mode since it can show the state to users.
 * To improve security of out application we should only run logger in development mode
 */
if (process.env.NODE_ENV === "development") {
  middlewares.push(logger);
}
export const store = createStore(rootReducer, applyMiddleware(...middlewares));
