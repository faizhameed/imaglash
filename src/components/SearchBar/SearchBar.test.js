import { shallow } from "enzyme";
import React from "react";

import SearchBar from "./SearchBar";
import { Provider } from "react-redux";
import { store } from "../../redux/store";

describe("SearchBar Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Provider store={store}>
        <SearchBar />
      </Provider>
    );
  });
  it("expect to render Search Bar", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
