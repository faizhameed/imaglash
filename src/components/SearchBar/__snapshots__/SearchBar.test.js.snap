// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component expect to render Search Bar 1`] = `ShallowWrapper {}`;
